### PR TITLE
WIP: Changed oauthclientauthorization name per Trello

### PR DESCRIPTION
**Pull request by mburke5678**
https://trello.com/c/2UC7OSgV/715-document-security-user-management-of-oauthclientauthorization
https://jira.coreos.com/browse/AUTH-30

Per Mo Kahn: It is currently not high enough priority to be worked on. Earliest I expect to see anything in 3.13.

### CODE REVIEW

**Comment by enj** [comment]
The naming change has not been API approved yet, so the docs should not be changed yet.

**Comment by mburke5678** [comment]
@enj Can you PTAL. I brought down your text from https://github.com/openshift/origin/pull/11909 and worked it into our existing topic, as you suggested. 
Will this functionality be available in the Web Console or just the CLI?

File: architecture/additional_concepts/authentication.adoc [hunk 1]
```diff
@@ -685,3 +685,186 @@ responses from `/oauth/authorize`, and how they should be handled:
 * `openshift_auth_form_password_count` counts the number of web console login attempts.
 * `openshift_auth_form_password_count_result` counts the number of web console login attempts by result (success or error).
 * `openshift_auth_password_total` counts the total number of `oc login` and web console login attempts.
+
+[[authentication-manage-oauth]]
+== Managing OAuth Objects
+
+Each user has an individual directory where OAuth Client Authorizations for that user are stored:
+*_/openshift.io/oauth/clientauthorizations/<userName>/<clientName>_*.
```
Comment (enj) [inline architecture/additional_concepts/authentication.adoc]: This is an implementation detail that will not be exposed to end users.

File: architecture/additional_concepts/authentication.adoc [hunk 2]
```diff
@@ -685,3 +685,186 @@ responses from `/oauth/authorize`, and how they should be handled:
...
+
+Having a specific directory allows for faster lookup times for a user. 
+Previously, {product-title} maintained OAuth Client Authorizations in etcd.
+
+The naming format for OAuth Client Authorizations has also changed. The new format is `<userName>::<clientName>`,
+where the previous format is `<userName>:<clientName>`.
```
Comment (enj) [inline architecture/additional_concepts/authentication.adoc]: Users never knew the old format, so the new format is not relevant to them.  Thus this information would go into the "admin" section.

File: architecture/additional_concepts/authentication.adoc [hunk 3]
```diff
@@ -685,3 +685,186 @@ responses from `/oauth/authorize`, and how they should be handled:
...
 * `openshift_auth_form_password_count_result` counts the number of web console login attempts by result (success or error).
 * `openshift_auth_password_total` counts the total number of `oc login` and web console login attempts.
+
+[[authentication-manage-oauth]]
+== Managing OAuth Objects
+
+Each user has an individual directory where OAuth Client Authorizations for that user are stored:
+*_/openshift.io/oauth/clientauthorizations/<userName>/<clientName>_*.
+
+Having a specific directory allows for faster lookup times for a user.
```
Comment (enj) [inline architecture/additional_concepts/authentication.adoc]: Implementation detail.

File: architecture/additional_concepts/authentication.adoc [hunk 4]
```diff
@@ -685,3 +685,186 @@ responses from `/oauth/authorize`, and how they should be handled:
...
+
+[[authentication-manage-oauth-cleanup]]
+=== Cleanup Old Objects
+
+It is not possible to change an exsiting Oauth Client Authorizations.
+
+Use the following command to delete all invalid Oauth Client Authorizations:
+
+----
+oc get oauthclientauthorization -o jsonpath="{range .items[*]}{.metadata.name}{\"\\n\"}{end}" | grep -v '::' | xargs -n 1 echo oc delete oauthclientauthorization
```
Comment (enj) [inline architecture/additional_concepts/authentication.adoc]: Commands that delete data are dangerous and not something I want to include in the docs.  I am ok with saying the "cluster admin has to delete...", but not ok with giving people copy paste logic that could harm their cluster.

File: architecture/additional_concepts/authentication.adoc [hunk 5]
```diff
@@ -685,3 +685,186 @@ responses from `/oauth/authorize`, and how they should be handled:
...
 * `openshift_auth_password_total` counts the total number of `oc login` and web console login attempts.
+
+[[authentication-manage-oauth]]
+== Managing OAuth Objects
+
+Each user has an individual directory where OAuth Client Authorizations for that user are stored:
+*_/openshift.io/oauth/clientauthorizations/<userName>/<clientName>_*.
+
+Having a specific directory allows for faster lookup times for a user. 
+Previously, {product-title} maintained OAuth Client Authorizations in etcd.
```
Comment (enj) [inline architecture/additional_concepts/authentication.adoc]: Umm, everything is still in etcd.  Nothing is stored locally or in some file system.

File: architecture/additional_concepts/authentication.adoc [hunk 6]
```diff
@@ -685,3 +685,186 @@ responses from `/oauth/authorize`, and how they should be handled:
 * `openshift_auth_form_password_count` counts the number of web console login attempts.
 * `openshift_auth_form_password_count_result` counts the number of web console login attempts by result (success or error).
 * `openshift_auth_password_total` counts the total number of `oc login` and web console login attempts.
+
+[[authentication-manage-oauth]]
+== Managing OAuth Objects
```
Comment (enj) [inline architecture/additional_concepts/authentication.adoc]: client authorizations

File: architecture/additional_concepts/authentication.adoc [hunk 7]
```diff
@@ -685,3 +685,186 @@ responses from `/oauth/authorize`, and how they should be handled:
...
+  scopes:
+  - user:info
+  userName: developer
+  userUID: cf29b7cb-b0ca-11e6-ac48-507b9dac97ff
+kind: List
+metadata: {}
+----
+<1> Each OAuth Client Authorization object is listed, starting with `apiVersion`.
+<2> Metadata added to identify the object.
+<3> The scope of the access request.
```
Comment (enj) [inline architecture/additional_concepts/authentication.adoc]: The list of authorized scopes from the given user to the specified client

File: architecture/additional_concepts/authentication.adoc [hunk 8]
```diff
@@ -685,3 +685,186 @@ responses from `/oauth/authorize`, and how they should be handled:
...
+Use the following command to delete all invalid Oauth Client Authorizations:
+
+----
+oc get oauthclientauthorization -o jsonpath="{range .items[*]}{.metadata.name}{\"\\n\"}{end}" | grep -v '::' | xargs -n 1 echo oc delete oauthclientauthorization
+----
+
+Use the following command to delete specific invalid Oauth Client Authorizations:
+
+----
+oc delete oauthclientauthorization <file-client-name>
```
Comment (enj) [inline architecture/additional_concepts/authentication.adoc]: ```<file-client-name>```?  where did file come from?

**Review by enj — CHANGES_REQUESTED** [review]

**Comment by enj** [comment]
@mburke5678 lets wait on any further doc changes until after the changes are actually added to origin (3.10+ timeframe).  As a start only the CLI will support this feature, but there is no reason the web console could not support it at some future point (it is an advanced use case so it may be too much for the console).

**Comment by mburke5678** [comment]
@enj Is this feature officially scheduled for 3.10? 

cc. @vikram-redhat

**Comment by enj** [comment]
No we had no cycles in 3.10 to work on this.

**Comment by vikram-redhat** [comment]
@mburke5678 ok - I have changed the label to 3.11.

**Comment by vikram-redhat** [comment]
@mburke5678 @enj - is this scheduled for 3.11?

**Comment by mburke5678** [comment]
@vikram-redhat Asked Mo via a note in Dev Trello for status

**Comment by mburke5678** [comment]
@vikram-redhat Per Mo Kahn: [It is currently not high enough priority to be worked on. Earliest I expect to see anything in 3.13](https://trello.com/c/Tcabgp9F/21-8-user-management-of-oauthclientauthorization#comment-5b607d3d7142130c6411747c).

**Pull request closed without merging**